src/FsAutoComplete.Core/Commands.fs

```diff
@@ -1120,3 +1120,28 @@ type Commands (serialize : Serializer, backgroundServiceEnabled) =
     }
 
     member x.GetChecker () = checker.GetFSharpChecker()
+
+    member x.ScopesForFile (file: string) = async {
+        let rec blockForCheckResults file opts = async {
+            match checker.TryGetRecentCheckResultsForFile(file, opts) with
+            | None -> 
+                do! Async.Sleep 100
+                return! blockForCheckResults file opts
+            | Some tyRes -> 
+                return tyRes
+        }
+        let file = Path.GetFullPath file
+        match state.TryGetFileCheckerOptionsWithLines file with
+        | Error s -> return Error s
+        | Ok (opts, source) -> 
+            do! Async.Sleep 5000
```

Comment:
I need a way to block this foldingRange request for a particular file until after typechecking/parsing, because the provider doesn't seem to be called again after errors...